tag-historian/nitaghistorian.yml

```diff
@@ -0,0 +1,928 @@
+        example: TagHistorianAsyncQueryCompleteRoutedMessage
+      queryJobs:
+        description: A list containing the information about each started query.
+        type: array
+        items:
+          $ref: '#/definitions/HistoricalTagAsyncQueryJob'
+  WebSocketSubscribeRequest:
+    title: Web Socket Subscribe Request
+    description: A web socket message sent by the client to subscribe to notifications when one or more tags have recorded new historical values. The server will send a *WebSocketSubscribeResponse* message to confirm the subscription and a *WebSocketAsyncValuesProcessed* message to notify of new values. Clients may send a *WebSocketUnsubscribeRequest* message to stop future notifications without closing the connection.
+    type: object
+    required:
+      - action
+      - searchPaths
+    properties:
+      action:
+        description: The action to perform. Must be set to "TagHistorianAsyncSubscribeRequest".
+        type: string
+        enum: ['TagHistorianAsyncSubscribeRequest']
+        example: TagHistorianAsyncSubscribeRequest
+      searchPaths:
+        description: Contains one or more tag paths to subscribe to. The paths may contain wildcards to subscribe to similar tags.
+        type: array
+        items:
+          type: string
+  WebSocketSubscribeResponse:
+    title: Web Socket Subscribe Response
+    description: A web socket message sent by the server to confirm a *WebSocketSubscribeRequest* sent by the client. A *WebSocketAsyncValuesProcessed* message will be sent each time new values are available.
+    type: object
+    properties:
+      action:
+        description: The action performed. Must be set to "TagHistorianAsyncSubscribeResponse".
+        type: string
+        enum: ['TagHistorianAsyncSubscribeResponse']
+        example: TagHistorianAsyncSubscribeResponse
+  WebSocketAsyncValuesProcessed:
+    title: Web Socket Async Values Processed
+    description: A web socket message sent by the server when new historical values are available for any of the tags the client has subscribed to via a *WebSocketSubscribeRequest* message.
+    type: object
+    properties:
+      action:
+        description: The action performed. Must be set to "TagHistorianAsyncValuesProcessedRoutedMessage".
+        type: string
+        enum: ['TagHistorianAsyncValuesProcessedRoutedMessage']
+        example: TagHistorianAsyncValuesProcessedRoutedMessage
+      values:
+        description: The newly recorded historical values.
+        type: array
+        items:
+          $ref: '#/definitions/HistoricalValue'
+  WebSocketUnsubscribeRequest:
+    title: Web Socket Unsubscribe Request
+    description: A web socket message sent by the client to unsubscribe from notifications when one or more tags have recorded new historical values created by a previous *WebSocketSubscribeRequest*. The server will respond with a *WebSocketUnsubscribeResponse* message to confirm.
```

Comment:
This string is a little long, but maybe that's just my laptop. I'd been setting a 110 character column guide for the swagger docs I've worked on so far.

Reply:
Yeah, the Swagger editor automatically wraps in the editor but not in the file, so it's easy to forget.